### PR TITLE
[VEN-1607] Peckshield Audit Findings Fix

### PR DESCRIPTION
**Pull request by web3rover**
## Description

Resolves #VEN-1607

## Checklist

- [x] I have updated the documentation to account for the changes in the code.
- [x] If I added new functionality, I added tests covering it.
- [x] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [x] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/VToken.sol [hunk 1]
```diff
@@ -881,17 +882,17 @@ contract VToken is
...
 
             uint256 _redeemAmount = mul_(redeemTokens, exchangeRate);
             if (_redeemAmount != 0 && _redeemAmount != redeemAmountIn) redeemTokens++; // round up
-            redeemAmount = redeemAmountIn;
+            redeemAmount = mul_ScalarTruncate(exchangeRate, redeemTokens);
         }
 
         // Revert if tokens is zero and amount is nonzero or token is nonzero and amount is zero
-        if ((redeemTokens == 0 && redeemAmount > 0) || (redeemTokens != 0 && redeemAmount == 0)) {
+        if ((redeemTokens == 0 && redeemAmount == 0) || (redeemTokens != 0 && redeemAmount == 0)) {
```
[the remark this answers is not in the document]
Comment (chechu) [reply contracts/VToken.sol:895]: In that case, we can simplify the `if` condition, reducing it to `if (redeemAmount == 0)`. And we should update the comments and the revert message

**Review by chechu — DISMISSED** [review]

**Review by defcon022 — DISMISSED** [review]

File: contracts/VToken.sol [hunk 2]
```diff
@@ -881,18 +882,18 @@ contract VToken is
...
              *  redeemTokens = redeemAmountIn / exchangeRate
-             *  redeemAmount = redeemAmountIn
+             *  redeemAmount = exchangeRate * redeemTokens
              */
             redeemTokens = div_(redeemAmountIn, exchangeRate);
 
             uint256 _redeemAmount = mul_(redeemTokens, exchangeRate);
             if (_redeemAmount != 0 && _redeemAmount != redeemAmountIn) redeemTokens++; // round up
-            redeemAmount = redeemAmountIn;
+            redeemAmount = mul_ScalarTruncate(exchangeRate, redeemTokens);
```
Comment (defcon022) [inline contracts/VToken.sol:891]: we can move this line and [this](https://github.com/VenusProtocol/isolated-pools/pull/247/files#diff-74967185a2c5eedfa0a1e0b396c10b82321afb533d1fca50ca5d896e81915e17R880) outside the if and else block as it is common.

File: contracts/VToken.sol [hunk 3]
```diff
@@ -868,31 +868,32 @@ contract VToken is
...
+
         /* If redeemTokensIn > 0: */
         if (redeemTokensIn > 0) {
             /*
              * We calculate the exchange rate and the amount of underlying to be redeemed:
              *  redeemTokens = redeemTokensIn
-             *  redeemAmount = redeemTokensIn x exchangeRateCurrent
              */
             redeemTokens = redeemTokensIn;
             redeemAmount = mul_ScalarTruncate(exchangeRate, redeemTokensIn);
```
Comment (defcon022) [inline contracts/VToken.sol:879]: We should remove this line from here, as we move it outside the if else block at [line#892](https://github.com/VenusProtocol/isolated-pools/pull/247/files#diff-74967185a2c5eedfa0a1e0b396c10b82321afb533d1fca50ca5d896e81915e17R892) @narayanprusty

File: contracts/VToken.sol (hunk 2 above)
Comment (defcon022) [reply contracts/VToken.sol]: https://github.com/VenusProtocol/isolated-pools/pull/247#discussion_r1229460347

**Review by chechu — APPROVED** [review]

**Review by defcon022 — APPROVED** [review]

**Pull request merged**